test/fail_compilation/fail16001.d

```diff
@@ -2,7 +2,7 @@
 /*
 TEST_OUTPUT:
 ---
-fail_compilation/fail16001.d(10): Deprecation: `(args) => { ... }` is a lambda that returns a delegate, not a multi-line lambda.
+fail_compilation/fail16001.d(10): Deprecation: Using `(args) => { ... }` is an ambiguous way to create a lambda that returns a delegate.
```

Comment:
This is just wrong and it doesn't tell the user anything.

Reply:
The message I wrote is similar to a bug report: explain what you saw as well as what you expected to see. The idea is the user wrote that expecting to see a mutli line delegate. So the message specifically says that is not what happened and briefly describes what actually happened instead.

Then the next line explains to them how to fix it for either intent.

This is most likely going to be seen by new users and meeting them where they are will hopefully cut down on the support requests (though I've also seen many times when people copy paste perfectly clear messages, they just never bothered actually reading it.....)

Reply:
> The idea is the user wrote that expecting to see a mutli line delegate

And that's where the message is lacking. It should not only cater to those making a mistake, but also to those actually using it correctly. As a matter of fact, since we're introducing a deprecation, it's quite likely that it will be seen by seasoned users first, before it is seen by newcomers...

The deprecation message should tell the users why what (s)he is doing is not possible, not state what the user (may have) intended to do. To draw a parallel with error messages, consider this:
```
void foo (string a);
void main ()
{
    foo(42);
}
```
Pick the better error message:
```
foo.d(4): Error: Function `void foo(string a)` is being called with arguments `(42)`
- OR -
foo.d(4): Error: cannot pass argument `42` of type `int` to parameter `string a`
```

I chose "ambiguous" for lack of a better word. Implicit string concatenation uses "error-prone", perhaps we should use that (disclaimer: I picked "error-prone") ?